packages/atomic/src/utils/initialization-utils.tsx

```diff
@@ -290,4 +290,26 @@ export function DeferUntilRender() {
   };
 }
 
+export function RemainHiddenUntilDescendantsRendered() {
+  return (component: ComponentInterface, methodName: string) => {
+    const {componentDidRender} = component;
+
+    component[methodName] = function () {
+      const element = getElement(this);
+      if (!element.classList.contains('atomic-hidden')) {
+        return;
+      }
+      element.style.visibility = 'hidden';
+      element.style.position = 'absolute';
+    };
+
+    component.componentDidRender = function () {
```

Comment:
with these decorators, I'm just wondering if setting lifecycle methods like this isn't dangerous, and it might override stuff defined in the component or even in other decorators?
or is that what `return componentDidRender && componentDidRender.call(this);` does?

Reply:
> or is that what return componentDidRender && componentDidRender.call(this); does?

Yes, it "overrides" them but then calls the original. The biggest risk is that the order in which these are applied is ambiguous.